File: lib/models/stats_model.dart
class ScreeningSummary {
  final int total;
  final int high;
  final int medium;
  final int low;

  ScreeningSummary({
    required this.total,
    required this.high,
    required this.medium,
    required this.low,
  });

  factory ScreeningSummary.fromJson(Map<String, dynamic> json) {
    return ScreeningSummary(
      total: int.tryParse(json['total'].toString()) ?? 0,
      high: int.tryParse(json['high'].toString()) ?? 0,
      medium: int.tryParse(json['medium'].toString()) ?? 0,
      low: int.tryParse(json['low'].toString()) ?? 0,
    );
  }
}

class ScreeningDetail {
  final int id;
  final String userName;
  final String childName;
  final String riskLevel;
  final int totalScore;
  final String createdAt;

  ScreeningDetail({
    required this.id,
    required this.userName,
    required this.childName,
    required this.riskLevel,
    required this.totalScore,
    required this.createdAt,
  });

  factory ScreeningDetail.fromJson(Map<String, dynamic> json) {
    return ScreeningDetail(
      id: json['id'],
      userName: json['user_name'] ?? '',
      childName: json['child_name'] ?? '',
      riskLevel: json['risk_level'] ?? '',
      totalScore: json['total_score'] ?? 0,
      createdAt: json['created_at'] ?? '',
    );
  }
}

class ScreeningStats {
  final ScreeningSummary summary;
  final List<ScreeningDetail> details;

  ScreeningStats({required this.summary, required this.details});

  factory ScreeningStats.fromJson(Map<String, dynamic> json) {
    return ScreeningStats(
      summary: ScreeningSummary.fromJson(json['summary']),
      details: (json['details'] as List)
          .map((e) => ScreeningDetail.fromJson(e))
          .toList(),
    );
  }
}

class UserStat {
  final int id;
  final String fullName;
  final String email;
  final int totalChildren;
  final int totalScreenings;
  final String? lastScreening;

  UserStat({
    required this.id,
    required this.fullName,
    required this.email,
    required this.totalChildren,
    required this.totalScreenings,
    this.lastScreening,
  });

  factory UserStat.fromJson(Map<String, dynamic> json) {
    return UserStat(
      id: json['id'],
      fullName: json['full_name'] ?? '',
      email: json['email'] ?? '',
      totalChildren: int.tryParse(json['total_children'].toString()) ?? 0,
      totalScreenings:
      int.tryParse(json['total_screenings'].toString()) ?? 0,
      lastScreening: json['last_screening'],
    );
  }
}

File: ai_child_backend/src/models/stats.model.js
const db = require('../config/db');

exports.getScreeningStats = async (from, to) => {
  let dateClause = '';
  const params = [];

  if (from && to) {
    dateClause = 'AND DATE(s.created_at) BETWEEN ? AND ?';
    params.push(from, to);
  } else if (from) {
    dateClause = 'AND DATE(s.created_at) >= ?';
    params.push(from);
  } else if (to) {
    dateClause = 'AND DATE(s.created_at) <= ?';
    params.push(to);
  }

  const [rows] = await db.execute(
    `SELECT
       COUNT(*) AS total,
       SUM(CASE WHEN s.risk_level = 'High' THEN 1 ELSE 0 END) AS high,
       SUM(CASE WHEN s.risk_level = 'Medium' THEN 1 ELSE 0 END) AS medium,
       SUM(CASE WHEN s.risk_level = 'Low' THEN 1 ELSE 0 END) AS low
     FROM mchat_sessions s
     JOIN users u ON s.user_id = u.id
     JOIN children c ON s.child_id = c.id
     WHERE u.role = 'USER'
     AND s.risk_level IS NOT NULL
     ${dateClause}`,
    [...params]
  );

  const [details] = await db.execute(
    `SELECT
       s.id,
       u.full_name AS user_name,
       c.full_name AS child_name,
       s.risk_level,
       s.total_score,
       s.created_at
     FROM mchat_sessions s
     JOIN users u ON s.user_id = u.id
     JOIN children c ON s.child_id = c.id
     WHERE u.role = 'USER'
     AND s.risk_level IS NOT NULL
     ${dateClause}
     ORDER BY s.created_at DESC`,
    [...params]
  );

  return {
    summary: rows[0],
    details,
  };
};

exports.getUserStats = async (sort) => {
  const orderMap = {
    count: 'total_screenings DESC',
    latest: 'last_screening DESC',
    name: 'u.full_name ASC',
  };

  const orderBy = orderMap[sort] || orderMap.count;

  const [rows] = await db.execute(
    `SELECT
       u.id,
       u.full_name,
       u.email,
       COUNT(DISTINCT c.id) AS total_children,
       COUNT(DISTINCT s.id) AS total_screenings,
       MAX(s.created_at) AS last_screening
     FROM users u
     LEFT JOIN children c ON c.user_id = u.id
     LEFT JOIN mchat_sessions s ON s.user_id = u.id AND s.risk_level IS NOT NULL
     WHERE u.role = 'USER'
     GROUP BY u.id, u.full_name, u.email
     ORDER BY ${orderBy}`
  );

  return rows;
};